test/ui/QuerySummaryTest.ts

```diff
@@ -156,5 +202,121 @@ export function QuerySummaryTest() {
         expect($$(test.cmp.element).text()).not.toEqual(jasmine.stringMatching(/^Results.*of.*/));
       });
     });
+
+    describe('when the option enableNoResultsFoundMessage is set to true', () => {
+      it(`when the noResultsFoundMessage is not specified
+          it should return the default message`, () => {
+        test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+          enableNoResultsFoundMessage: true
+        });
+        expect(test.cmp.options.noResultsFoundMessage).toBe(`No results for ${queryTag}`);
+      });
+
+      it(`when the noResultsFoundMessage is specified 
+          it should return the custom message`, () => {
+        test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+          enableNoResultsFoundMessage: true,
+          noResultsFoundMessage: 'customMessage'
+        });
+        expect(test.cmp.options.noResultsFoundMessage).toBe('customMessage');
+      });
+
+      describe('when the noResultsFoundMessage is parsed', () => {
+        it(`when there is no query tag in the noResultsFoundMessage
+            it should not change the initial message`, () => {
+          test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: 'customMessage'
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
```

Comment:
This is repeated often in that suite of test. Add an utility function for this in that describe block.

Reply:
No problem good point I will add the utility function!